traffic_editor/gui/editor.cpp

```diff
@@ -1551,9 +1593,37 @@ void Editor::property_editor_cell_changed(int row, int column)
     if (name == "name")
       v.name = value;
     else if (name == "x (pixels)")
+    {
+      if (auto& impl = project.building.levels[level_idx].crowd_sim_impl)
+      {
+        auto& agent_groups = impl->get_agent_groups();
+        for (auto& agent_group:agent_groups)
+        {
+          auto p = agent_group.get_spawn_point();
+          if (p.first == v.x)
+          {
+            agent_group.set_spawn_point(stod(value), p.second);
+          }
+        }
+      }
       v.x = stof(value);
+    }
     else if (name == "y (pixels)")
+    {
+      if (auto& impl = project.building.levels[level_idx].crowd_sim_impl)
+      {
+        auto& agent_groups = impl->get_agent_groups();
+        for (auto& agent_group:agent_groups)
+        {
+          auto p = agent_group.get_spawn_point();
+          if (p.second == v.y)
+          {
+            agent_group.set_spawn_point(p.first, stod(value));
+          }
+        }
+      }
```

Comment:
I'd try to refactor all those chunks into crowdsim and make the changes in editor.cpp as small as possible. They went from being one liners to being ~20 lines each and I think that affects readability quite a bit (this file is already very bulky).